test/unit/math/gpu/opencl_matrix_test.cpp

```diff
@@ -1,12 +1,40 @@
 #ifdef STAN_OPENCL
 #include <stan/math/prim/mat.hpp>
+#include <stan/math/rev/mat.hpp>
+#include <test/unit/math/rev/mat/fun/util.hpp>
 #include <stan/math/gpu/opencl_context.hpp>
 #include <stan/math/gpu/matrix_gpu.hpp>
 #include <gtest/gtest.h>
 #include <algorithm>
 #include <vector>
 
-TEST(MathMatrixGPU, matrix_gpu_creation) {
+void test_cache_speed() {
+  auto m = stan::math::matrix_d::Random(100, 100).eval();
+  std::chrono::steady_clock::time_point begin1
```

Comment:
Timing unit tests like this are not a good idea if they can be avoided because they often fail to reproduce spuriously on different kinds of hardware. It's better to either expose more functionality for testing or find other ways to test it. Here, I might suggest a few things:
1) Create an eigen matrix, do an operation, save the result, change the eigen matrix, perform the operation again and check against first result. Should be the same even though the matrix is different now. This is nice because it tests the behavior you expect and highlights this sort of unintuitive behavior at the same time.
2) Attempt to provide your own instrumented implementation of one of the internal operations that you want to check - in this case it might be something in opencl, and it might not be possible to mock it out or instrument it. But if you can find existing instrumentation in opencl or a nice way to mock in C++ this is great.
3) find other observable aspects of the new behavior to test!

Reply:
Yeah after re-reading, checking host->device and device->device times doesn't make a ton of sense anyway. 

Will def do one and three, need to think about two